File: Day4/project.py
print('''
*******************************************************************************
          |                   |                  |                     |
 _________|________________.=""_;=.______________|_____________________|_______
|                   |  ,-"_,=""     `"=.|                  |
|___________________|__"=._o`"-._        `"=.______________|___________________
          |                `"=._o`"=._      _`"=._                     |
 _________|_____________________:=._o "=._."_.-="'"=.__________________|_______
|                   |    __.--" , ; `"=._o." ,-"""-._ ".   |
|___________________|_._"  ,. .` ` `` ,  `"-._"-._   ". '__|___________________
          |           |o`"=._` , "` `; .". ,  "-._"-._; ;              |
 _________|___________| ;`-.o`"=._; ." ` '`."\` . "-._ /_______________|_______
|                   | |o;    `"-.o`"=._``  '` " ,__.--o;   |
|___________________|_| ;     (#) `-.o `"=.`_.--"_o.-; ;___|___________________
____/______/______/___|o;._    "      `".o|o_.--"    ;o;____/______/______/____
/______/______/______/_"=._o--._        ; | ;        ; ;/______/______/______/_
____/______/______/______/__"=._o--._   ;o|o;     _._;o;____/______/______/____
/______/______/______/______/____"=._o._; | ;_.--"o.--"_/______/______/______/_
____/______/______/______/______/_____"=.o|o_.--""___/______/______/______/____
/______/______/______/______/______/______/______/______/______/______/______/ 
*******************************************************************************
''')
print('Welcome to Treasure Island,\n')
print('Your mission is to find the treasure\n')

choice1 = input("Where would you like to go? Type 'left' or 'right'\n").lower()

if choice1 == 'left':
    choice2 = input("You find a river blocking your way. Do you want to 'swim' across it or 'wait' a little bit?\n")
    if choice2 == 'wait':
        choice3 = input("As you were waiting, suddenly 3 doors popped out from the ground. Which door would you like to enter? 'Red' 'Blue' 'Yellow'\n")
        if choice3 == 'red':
            print("You were burned by fire. Game Over\n")
        elif choice3 == 'yellow':
            print("Congratulations. You find the treasure! You WIN!!\n")
        elif choice3 == 'blue':
            print("Dangerous beast already waiting inside the door. Game Over\n")
        else:
            print("You died without knowing why you died. Game Over\n")
    else:
        print('Attacked by trout. Game Over\n')
else:
    print('Fall into Hole. Game Over\n')

File: Day4/project.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cctype>

int main() {
    std::cout<<"Welcome to Treasure Island,\n";
    std::cout<<"Your mission is to find the treasure\n";

    std::string choice1;
    std::cout<<"Where would you like to go? Type 'left' or 'right'\n";
    std::cin>>choice1;
    std::transform(choice1.begin(), choice1.end(), choice1.begin(), [](unsigned char c) -> unsigned char {return std::tolower(c);});

    if (choice1 == "left"){
        std::string choice2;
        std::cout<<"You find a river blocking your way. Do you want to 'swim' across it or 'wait' a little bit?\n";
        std::cin>>choice2;
        std::transform(choice2.begin(), choice2.end(), choice2.begin(), [](unsigned char c) -> unsigned char {return std::tolower(c);});
        if(choice2 == "wait"){
            std::string choice3;
            std::cout<<"As you were waiting, suddenly 3 doors popped out from the ground. Which door would you like to enter? 'Red' 'Blue' 'Yellow'\n";
            std::cin>>choice3;
            std::transform(choice3.begin(), choice3.end(), choice3.begin(), [](unsigned char c) -> unsigned char {return std::tolower(c);});
            if(choice3 == "red"){
                std::cout<<"You were burned by fire. Game Over\n";
            } else if (choice3 == "blue"){
                std::cout<<"Dangerous beast already waiting inside the door. Game Over\n";
            } else if (choice3 == "yellow"){
                std::cout<<"Congratulations. You find the treasure! You WIN!!\n";
            }else {
                std::cout<<"You died without knowing why you died. Game Over\n";
            }
        } else {
            std::cout<<"Attacked by trout. Game Over\n";
        }
    } else {
        std::cout<<"Fall into Hole. Game Over\n";
    }
}